src/common/utility.js

```diff
@@ -244,3 +244,29 @@ export const redirectOpenLiveChatBox = (is_redirect) => {
         navigate(live_chat_redirection_link, { replace: true })
     }
 }
+
+// Function which returns sub path to the specific trading platform
+export const redirectToTradingPlatform = () => {
+    const dmt5_url = 'mt5'
+    const dbot_url = 'bot'
+    const derivx_url = 'derivx'
+
+    const isUserAt = () => {
+        const checkURL = (platform) => window.location.pathname.includes(platform)
+        return { dmt5: checkURL(dmt5_url), dbot: checkURL(dbot_url), derivx: checkURL(derivx_url) }
+    }
+
+    const redirectTo = () => {
+        if (isUserAt().dmt5) {
+            return dmt5_url
+        } else if (isUserAt().dbot) {
+            return dbot_url
+        } else if (isUserAt().derivx) {
+            return derivx_url
+        } else {
+            return ''
+        }
+    }
```

Comment:
Can we loop through the object/array and return its entries so that it's less repetitive instead?

Reply:
Yeah, this is how I refactored the codes:

```javascript
const supported_platforms = ['mt5', 'bot', 'derivx']
export const redirectToTradingPlatform = () =>
    supported_platforms.filter(
        (platform, index) =>
            window.location.pathname.includes(platform) && supported_platforms[index],
    )
```

Let me know if it's okay!